src/core/ports.rs

```diff
@@ -1,18 +1,33 @@
 use std::collections::HashMap;
 
+#[derive(Debug, PartialEq, Copy, Clone, Eq, Hash)]
+struct Sctp(u16);
+#[derive(Debug, PartialEq, Copy, Clone, Eq, Hash)]
+struct Tcp(u16);
+#[derive(Debug, PartialEq, Copy, Clone, Eq, Hash)]
+struct Udp(u16);
+
 /// The exposed ports of a running container.
 #[derive(Debug, PartialEq, Default)]
 pub struct Ports {
-    mapping: HashMap<u16, u16>,
+    tcp: HashMap<Tcp, Tcp>,
+    udp: HashMap<Udp, Udp>,
+    sctp: HashMap<Sctp, Sctp>,
 }
 
 impl Ports {
     pub fn new(ports: HashMap<String, Option<Vec<HashMap<String, String>>>>) -> Self {
-        let mapping = ports
+        let tcp = HashMap::new();
+        let udp = HashMap::new();
+        let sctp = HashMap::new();
+
+        ports
             .into_iter()
             .filter_map(|(internal, external)| {
-                // internal is '8332/tcp', split off the protocol ...
-                let internal = internal.split('/').next()?;
+                // internal is ')8332/tcp', split off the protocol ...
```

Comment:
A typo got into this comment :)

```suggestion
                // internal is '8332/tcp', split off the protocol ...
```